master/internal/prom/prom.go

```diff
@@ -0,0 +1,233 @@
+package prom
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/device"
+)
+
+var (
+	containerIDToAllocationID = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "container_id_allocation_id",
+		Help: `
+Exposes mapping of allocation ID to container ID`,
+	}, []string{"container_id", "allocation_id"})
+
+	allocationIDToTaskID = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "allocation_id_task_id",
+		Help: `
+Exposes mapping of allocation ID to task ID`,
+	}, []string{"allocation_id", "task_id"})
+
+	containerIDToRuntimeID = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "container_id_runtime_container_id",
+		Help:      "a mapping of the container ID to the container ID given be the runtime",
+	}, []string{"container_runtime_id", "container_id"})
+
+	experimentIDToLabels = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "experiment_id_label",
+		Help:      "a mapping of the experiment ID to the labels",
+	}, []string{"experiment_id", "label"})
+
+	gpuUUIDToContainerID = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: "det",
+		Name:      "gpu_uuid_container_id",
+		Help: `
+Exposes mapping of Determined's container ID to the GPU UUID/device ID as given by nvidia-smi
+`,
+	}, []string{"gpu_uuid", "container_id"})
+
+	// DetStateMetrics is a prometheus registry containing all exported user-facing metrics.
+	DetStateMetrics = prometheus.NewRegistry()
+)
+
+const (
+	cAdvisorPort = ":8080"
+	dcgmPort     = ":9400"
+
+	// The are extra labels added to metrics on scrape.
+	detAgentIDLabel      = "det_agent_id"
+	detResourcePoolLabel = "det_resource_pool"
+
+	targetsFile = "/etc/determined/targets.json"
+)
+
+type fileSDConfigEntry struct {
+	Targets []string          `json:"targets"`
+	Labels  map[string]string `json:"labels"`
+}
+
+func init() { //nolint: gochecknoinits
+	DetStateMetrics.MustRegister(containerIDToAllocationID)
+	DetStateMetrics.MustRegister(containerIDToRuntimeID)
+	DetStateMetrics.MustRegister(gpuUUIDToContainerID)
+	DetStateMetrics.MustRegister(experimentIDToLabels)
+	DetStateMetrics.MustRegister(allocationIDToTaskID)
+}
+
+// AssociateAllocationContainer associates an allocation with its container ID.
+func AssociateAllocationContainer(aID string, cID string) {
```

Comment:
since it's so easy to mix these up when calling, i think it's better to drop the `string(x)`/`.String()` calls and let these take the more descriptive types, and then it can do the conversion itself.